design/commands.md

```diff
@@ -0,0 +1,201 @@
+# Commands (Sharing Binaries)
+
+## Context
+
+Currently, there's no clear way to share a binaries (or scripts) across multiple jobs.  Some people have chosen to share via git repositories, yum packages, node modules, and even docker images.  But nothing is standardized and works easily in all containers.
+
+## Status
+
+- 11 September 2017: Proposal submitted
+
+## Proposal
+
+Have a single interface for executing a versioned command (via remote binary, docker image, or habitat package) during a Screwdriver build.  Managed and discoverable through the Screwdriver interface.
+
+This provides a single abstraction for execution of shared code.
+
+```bash
+$ sd_cmd chefdk/knife@12 search node '*:*'
```

Comment:
We should make this `sd-cmd` instead of `sd_cmd` to be consistent with shared steps.

Reply:
This is to replace shared steps, so `sd-step` will be removed after this is implemented.  I prefer the `_` instead of `-` personally, although there is no limitation to using one vs. the other.